client/executor/wasmtime/src/runtime.rs

```diff
@@ -599,11 +601,13 @@ where
 /// [`create_runtime_from_artifact`] to get more details.
 unsafe fn do_create_runtime<H>(
 	code_supply_mode: CodeSupplyMode<'_>,
-	config: Config,
+	mut config: Config,
 ) -> std::result::Result<WasmtimeRuntime, WasmError>
 where
 	H: HostFunctions,
 {
+	replace_strategy_if_broken(&mut config.semantics.instantiation_strategy);
```

Comment:
Why not do this call in `common_config`?

Reply:
I considered it, but honestly that's kind of messy, since the `instantiation_strategy` is also used outside of `common_config`, so `common_config` would have to modify the `semantics` which is passed to it, and if it did that then instead of doing only one thing (creating a config for the engine) it'd be doing two things (creating a config *and* replacing the strategy), so I felt it'd be cleaner to just do it outside of `common_config` and have it done explicitly.